File: WC_25_qual/wc25c.py
#* we will calculate C(n+1,k+1) due to sum of series. n is the first modulo, k is the input
n = 93623149
k = int(input())
k += 1
mod = 1000000007

def fexp (a, b):
    if b == 0:
        return 1
    if b % 2 == 0:
        return fexp(a * a % mod, b // 2)
    return a * fexp(a, b - 1) % mod

def inv (a):
    return fexp(a, mod - 2)

nom = 1
denom = 1
r = min(k, n - k)

for i in range(r):
    nom = nom * (n - i) % mod
    denom = denom * (i + 1) % mod

res = nom * inv(denom) % mod
if k == 2:
    res += 2
elif k == 3:
    res += 1
print(res % mod)

File: WC_25_qual/wc25c.cpp
#include <iostream>
using namespace std;

const int MOD = 1000000007;

inline long long fexp(long long a, long long b) {
    long long res = 1;
    while (b > 0) {
        if (b % 2 == 1) {
            res = (res * a) % MOD;
        }
        a = (a * a) % MOD;
        b /= 2;
    }
    return res;
}

inline long long mod_inv(long long a) {
    return fexp(a, MOD - 2);
}

int main() {
    const long long n = 93623149;
    long long k;
    cin >> k;
    k += 1;
    k = min(k, n - k);
    long long res = 1;

    for (long long i = 0; i < k; ++i) {
        res = (res * (n - i) % MOD) * mod_inv(i + 1) % MOD; 
    }

    cout << res << endl;

    return 0;
}
